applications/vessel-history/src/features/home/Home.tsx

```diff
@@ -49,8 +51,15 @@ const Home: React.FC<LoaderProps> = (): React.ReactElement => {
 
   const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     dispatchQueryParams({ q: e.target.value })
+    setQuery(e.target.value)
   }
 
+  useEffect(() => {
+    if (query !== '') {
+      dispatch(fetchVesselSearchThunk({ query: query, offset: 0 }))
```

Comment:
When performing searches is not cancelling any ongoing search request, here is a proposal to fix it.  https://github.com/GlobalFishingWatch/frontend/pull/462

Reply:
mmmm I don't know, would be good only if is supported by the backend because is a expensive query. I prefer to keep the fetch because is cached and if you go back with your query you will have the results

Reply:
But the search api call is slow, and the UI is delaying even more the response to the user because it waits for all of the requests to be completed (they are processed in a queue, not in parallel). Normally when this scenario happens is because the user did not finished writing the search keyword (did a pause while writing) or because a mistake was made while typing; so I don't think caching the results for that 'intermediate' search would add any value.